src/microbe_stage/MicrobeHUD.cs

```diff
@@ -756,31 +757,86 @@ public void SendEditorButtonToTutorial(TutorialState tutorialState)
     /// </summary>
     private void UpdateNeededBars()
     {
-        var compounds = stage!.Player?.Compounds;
+        // if the player isn't in a colony than do everything the way its used to be done.
+        if (stage!.Player!.Colony == null)
+        {
+            var compounds = stage!.Player?.Compounds;
 
-        if (compounds?.HasAnyBeenSetUseful() != true)
-            return;
+            if (compounds?.HasAnyBeenSetUseful() != true)
+                return;
 
-        if (compounds.IsSpecificallySetUseful(oxytoxy))
-        {
-            agentsPanel.Show();
+            if (compounds.IsSpecificallySetUseful(oxytoxy))
+            {
+                agentsPanel.Show();
+            }
+            else
+            {
+                agentsPanel.Hide();
+            }
+
+            foreach (ProgressBar bar in compoundBars)
+            {
+                var compound = SimulationParameters.Instance.GetCompound(bar.Name);
+
+                if (compounds.IsUseful(compound))
+                {
+                    bar.Show();
+                }
+                else
+                {
+                    bar.Hide();
+                }
+            }
         }
         else
         {
-            agentsPanel.Hide();
-        }
+            // Stores the names of the Progressbars,
+            // then during the code if they are true they git to avoid flashing bars.
+            Dictionary<string, bool> barsOn = new Dictionary<string, bool>();
+            foreach (ProgressBar bar in compoundBars)
+            {
+                barsOn.Add(bar.Name, false);
+            }
 
-        foreach (ProgressBar bar in compoundBars)
-        {
-            var compound = SimulationParameters.Instance.GetCompound(bar.Name);
+            // If Oxytoxy is set to useful show then if every sell is
+            bool isOxytoxySetUseful = false;
+            foreach (var microbe in stage!.Player!.Colony!.ColonyMembers)
+            {
+                var compounds = microbe.Compounds;
+
+                if (compounds.IsSpecificallySetUseful(oxytoxy))
+                {
+                    agentsPanel.Show();
+                    isOxytoxySetUseful = true;
+                }
+
+                foreach (ProgressBar bar in compoundBars)
+                {
+                    var compound = SimulationParameters.Instance.GetCompound(bar.Name);
+
+                    if (compounds.IsUseful(compound))
+                    {
+                        barsOn[bar.Name] = true;
+                    }
+                }
+            }
```

Comment:
I think you should refactor this and the above method into a new method with a signature like: `UpdateBarVisibility(bool showAgents, Func<bool, Compound> isUseful)` (I may have gotten the func arguments backwards). But the basic logic then would be that these cases would compute the value for showAgents and pass a lambda for the isUseful parameter. This case would use a lambda like `(compound) => stage.Player.Colony.ColonyMembers.Any(c => c.Compounds.IsUseful(compound))`